TabClasses/Acquisition.py

```diff
@@ -0,0 +1,764 @@
+                            else:
+                                self.mainWindow.digitalPlots[i].setY(0)
+
+                # transceiver
+                if self.mainWindow.romode.value in [3, 4]:
+                    transceiverOffset = 0
+                    if self.mainWindow.romode.value == 4:
+                        nbitsPerDBit = self.dsamples
+                        if self.mainWindow.dsamples % 8 != 0:
+                            nbitsPerDBit += (8 - (self.dsamples % 8))
+                        transceiverOffset += self.mainWindow.nDbitEnabled * (nbitsPerDBit // 8)
+                    # print(f'transceiverOffset:{transceiverOffset}')
+                    trans_array = np.array(np.frombuffer(data, offset=transceiverOffset, dtype=np.uint16))
+                    for i in range(4):
+                        checkBox = getattr(self.mainWindow, f"checkBoxTransceiver{i}Plot")
+                        if checkBox.isChecked():
+                            waveform = np.zeros(self.tsamples * 4)
+                            for iSample in range(self.tsamples * 4):
+                                waveform[iSample] = trans_array[iSample * self.mainWindow.nTransceiverEnabled + i]
+                            self.mainWindow.transceiverPlots[i].setData(waveform)
+
+
+            # image
+            else:
+
+                # analog
+                if self.mainWindow.romode.value in [0, 2]:
+                    # get zoom state
+                    viewBox = self.mainWindow.plotAnalogImage.getView()
+                    state = viewBox.getState()
+
+                    # get histogram (colorbar) levels and histogram zoom range
+                    levels = self.mainWindow.plotAnalogImage.getHistogramWidget().item.getLevels()
+                    histRange = self.mainWindow.plotAnalogImage.getHistogramWidget().item.getHistogramRange()
+
+                    analog_array = np.array(
+                        np.frombuffer(data, dtype=np.uint16, count=self.mainWindow.nADCEnabled * self.asamples))
+
+                    try:
+                        self.mainWindow.analog_frame = decoder.decode(analog_array, self.mainWindow.pixelMapAnalog)
+                        self.mainWindow.plotAnalogImage.setImage(self.mainWindow.analog_frame.T)
+                    except Exception as e:
+                        self.mainWindow.statusbar.setStyleSheet("color:red")
+                        message = f'Warning: Invalid size for Analog Image. Expected {self.mainWindow.nAnalogRows * self.mainWindow.nAnalogCols} size, got {analog_array.size} instead.'
+                        self.updateCurrentFrame('Invalid Image')
+                        self.mainWindow.statusbar.showMessage(message)
+                        print(message)
+                        pass
+
+                    # keep the zoomed in state (not 1st image)
+                    if self.mainWindow.firstAnalogImage:
+                        self.firstAnalogImage = False
```

Comment:
This needs to be updated to: 
```python
self.mainWindow.firstAnalogImage = False
```
Otherwise we get a regression in the limits of the colorbar resetting after each acq, and also the initialisation of the variable needs to be updated. (or moved from CtbGUI)

Reply:
Yes that was an unintentional error. I missed it when trying to change the variable's occurrences.